### PR TITLE
feat: support custom http client

### PR DESCRIPTION
**Pull request by notdu**

### CODE REVIEW

File: casdoorsdk/base.go [hunk 1]
```diff
@@ -24,6 +24,27 @@ import (
 	"strings"
 )
 
+var (
+	// globalClient is a shared http Client.
+	globalClient = NewHTTPClient()
+)
+
+// SetCustomHTTPClient sets custom http Client.
+func SetCustomHTTPClient(s Doer) {
```
Comment (hsluoyz) [inline casdoorsdk/base.go:33]: Doer --> HttpClientInterface

File: casdoorsdk/base.go [hunk 2]
```diff
@@ -24,6 +24,27 @@ import (
 	"strings"
 )
 
+var (
+	// globalClient is a shared http Client.
+	globalClient = NewHTTPClient()
```
Comment (hsluoyz) [inline casdoorsdk/base.go:29]: globalClient --> client

File: casdoorsdk/base.go [hunk 3]
```diff
@@ -24,6 +24,27 @@ import (
...
+)
+
+// SetCustomHTTPClient sets custom http Client.
```
Comment (hsluoyz) [inline casdoorsdk/base.go:32]: All `HTTP` renamed to `Http`

File: casdoorsdk/base.go [hunk 4]
```diff
@@ -24,6 +24,27 @@ import (
...
+}
+
+// Doer interface has the method required to use a type as custom http client.
+// The net/*http.Client type satisfies this interface.
+type Doer interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// NewHTTPClient returns a new standard http Client.
+func NewHTTPClient() Doer {
```
Comment (hsluoyz) [inline casdoorsdk/base.go:44]: NewDefaultClient

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: casdoorsdk/base.go (hunk 1 above)
Comment (notdu) [reply casdoorsdk/base.go:33]: @hsluoyz It's already an interface. I think it should be `HttpClient`.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**